### DISCUSSION TITLE
Auto-fill from Bitwarden

### DISCUSSION THREAD

**Discussion by fiatguy85** [authelia/authelia#6322] [Q&A] [answered]
Hello, I cannot seem to get my Authelia page to auto-fill from the Bitwarden extension in Brave browser or Chromium.  I don't have an issue with this working on most websites.  Is this an Authelia setting or any hints on what the issue could be?

**Comment by james-d-elliott** [comment, marked as answer]
Best check is the dev console to see which security policy is blocking it (probably CSP/CORB/CORP/COEP headers). You can theoretically just remove the header if you wanted, however you negate several protection measures we recommend by default.

**Comment by fiatguy85** [comment]
I didn't have those headers set before.  I have learned about them and set them up correctly, but I'm not sure they are related to the issue.  

I also tried linking the field names that authelia uses inside Bitwarden for autofilling, but that was also unsuccessful.  I guess this has to be an issue with Bitwarden's Chrome Extension, so I will try to address it there.

**Reply by james-d-elliott** [reply]
It might be, it's possible an issue on our end however we do use conventional names and input types for the fields themselves and I think users have no issues with it in Firefox. If the browser dev tools console is not showing an error then it's likely an issue on their end however.

**Reply by fiatguy85** [reply]
I get the same behavior in Firefox and Chromium, with the Bitwarden extension.  When I switch the dev console to verbose, I got the following message in Chromium and Brave.  It looks like a recommendation to use a \<form\> element for the login.

[DOM] Password field is not contained in a form: (More info: https://goo.gl/9p2vKq)

**Reply by james-d-elliott** [reply]
We can try changing that but the autocomplete attribute is meant to govern this entirely according to the relevant specs at the time of implementation. It's possible it's changed.

**Reply by james-d-elliott** [reply]
I believe this is fixed in the latest build of v4.38.0-beta3. Please ensure you read the pinned discussion items in the [Pre-Release Feedback](https://github.com/authelia/authelia/discussions/categories/pre-release-feedback?discussions_q=is%3Aopen+category%3A%22Pre-Release+Feedback%22) discussion forum before trying.

**Reply by fiatguy85** [reply]
Ok, thank you for the update.  

I am running this on Archlinux from the release binary right now and I did not have much luck building a package from git source.  I will see if I can switch over to Docker and test this in the next few days.

**Reply by james-d-elliott** [reply]
Binaries are available here: https://buildkite.com/authelia/authelia/builds/25342#annotation-ctx-success

**Reply by fiatguy85** [reply]
Now, after messing around with things, auto-fill seems to be working for me, even in 4.37.5.  It may be, that your original suggestion was correct, and I fixed something in the headers that has it working correctly now.  For posterity, I will document the current headers here, as that may be the solution.  Not everything is necessarily relevant, as some may be required for other services.  

```
add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
add_header Content-Security-Policy "default-src 'self' https://mysite.com https://*.mysite.com/; frame-src https://mysite.com https://*.mysite.com; object-src 'none'; style-src 'self' https://*.mysite.com 'unsafe-inline'; frame-ancestors 'self' https://mysite.com https://*.mysite.com; base-uri 'self'" always;
add_header X-Frame-Options "SAMEORIGIN" always;
add_header X-Content-Type-Options "nosniff" always;
add_header Referrer-Policy "same-origin" always;
add_header Permissions-Policy "microphone=(), camera=()" always;
add_header Cross-Origin-Embedder-Policy "require-corp; report-to='default'" always;
add_header Cross-Origin-Opener-Policy "same-origin; report-to='default'" always;
add_header Cross-Origin-Resource-Policy "cross-origin" always;
```

I am not having much luck getting 4.38.0-beta3 running.  For some reason, I cannot get it to connect to my redis server.  I am not using a username/password or tls to connect to redis.  I tried IP:Port in addition to the socket connection, but I was not able to get the beta to connect to it.  I tried this in both Docker and with the binary, built the same way it is in the AUR.  

I'm not sure if it's a configuration issue somehow on my side, or if it's a different issue with the beta.  The socket connection is working for me with both Docker and the AUR-bin and the standard 4.37.5 release.

```
authelia  | time="2023-11-29T02:09:28Z" level=fatal msg="redis connection error: dial tcp: lookup /var/run/redis/redis.sock: no such host" stack="github.com/authelia/authelia/v4/internal/commands/context.go:147 (*CmdCtx).LoadProviders\ngithub.com/authelia/authelia/v4/internal/commands/root.go:71     (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.0/command.go:983                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.0/command.go:1115                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.0/command.go:1039                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\nruntime/internal/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1650                                         goexit"
```
Relevant configuration:

```
  redis:
    # host: 127.0.0.1
    # port: 6379
    host: '/var/run/redis/redis.sock'
    # username: authelia
    # password: session.redis.password
    database_index: 0
    maximum_active_connections: 8
    minimum_idle_connections: 0
    # tls:
```

**Reply by james-d-elliott** [reply]
Interesting, looks like it's trying to dail a tcp socket instead of a domain socket, I don't think anything changed here but I'll take a look.
